exrc-xdg.patch

```diff
@@ -0,0 +1,112 @@
+diff --git a/ex.c b/ex.c
+index a0ad9d8..26ccde9 100644
+--- a/ex.c
++++ b/ex.c
+@@ -22,6 +22,7 @@ int xpac;			/* print autocomplete options */
+ int xkwdcnt;			/* number of search kwd changes */
+ int xbufcur;			/* number of active buffers */
+ int xgrec;			/* global vi/ex recursion depth */
++int xexrc = 0;			/* read .exrc from the current directory */
+ struct buf *bufs;		/* main buffers */
+ struct buf tempbufs[2];		/* temporary buffers, for internal use */
+ struct buf *ex_buf;		/* current buffer */
+@@ -889,6 +890,7 @@ static struct option {
+ 	{"td", &xtd},
+ 	{"shape", &xshape},
+ 	{"order", &xorder},
++	{"exrc", &xexrc},
+ 	{"hl", &xhl},
+ 	{"hll", &xhll},
+ 	{"hlw", &xhlw},
+@@ -1218,6 +1220,51 @@ void ex(void)
+ 	xgrec--;
+ }
+ 
++void ex_script(FILE *fp)
++{
++	char done = 0;
++	do {
++		size_t n = 128, i = 0;
++		int c;
++		char *ln = malloc(128);
++		while ((c = fgetc(fp)) != EOF && c != '\n') {
++			if (i >= n - 2) {
++				n += 128;
++				ln = erealloc(ln, n);
++			}
++			ln[i++] = c;
++		}
++		if (!i) {
++			free(ln);
++			done = 1;
++			break;
++		}
++		ln[i] = '\0';
++		ex_command(ln);
++		free(ln);
++	} while(!done);
++}
++
++void load_exrc(char *exrc)
++{
++	struct stat st;
++	if (stat(exrc, &st) == 0) {
++		if (st.st_uid == getuid() && !(st.st_mode & S_IWGRP) && !(st.st_mode & S_IWOTH)) {
++			FILE *fp = fopen(exrc, "r");
++			if (fp) {
++				ex_script(fp);
++				fclose(fp);
++			} else {
++				fprintf(stderr, "Cannot open %s\n", exrc);
++				exit(EXIT_FAILURE);
++			}
++		} else {
++			fprintf(stderr, "Bad permissions on %s\n", exrc);
++			exit(EXIT_FAILURE);
++		}
++	}
++}
++
+ void ex_init(char **files, int n)
+ {
+ 	xbufsalloc = MAX(n, xbufsalloc);
+@@ -1229,6 +1276,20 @@ void ex_init(char **files, int n)
+ 	} while (--n > 0);
+ 	xmpt = 0;
+ 	xvis &= ~8;
+-	if ((s = getenv("EXINIT")))
++	if ((s = getenv("EXINIT"))) {
+ 		ex_command(s)
++	} else {
++		char *homeenv = getenv("HOME");
++		char *xdgconfighomeenv = getenv("XDG_CONFIG_HOME");
++		if (xdgconfighomeenv) {
```

Comment:
Standard XDG behavior is that if the XDG env is not set, use the standard fallback, here the fallback is `~/.config` IIRC.  Something like this
```c
char *xdgconfighomeenv = getenv("XDG_CONFIG_HOME");
if (xdgconfighomeenv) {
    char exrc[PATH_MAX];
    snprintf(exrc, sizeof(exrc), "%s/nextvi/exrc", xdgconfighomeenv);
    load_exrc(exrc);
} else if (homeenv) {
    char exrc[PATH_MAX];
    snprintf(exrc, sizeof(exrc), "%s/.config/nextvi/exrc", homeenv);
    load_exrc(exrc);
}
if (homeenv)
    char exrc[PATH_MAX];
    snprintf(exrc, sizeof(exrc), "%s/.exrc", homeenv);
    load_exrc(exrc);
}